controllers/webspherelibertyapplication_controller.go

```diff
@@ -235,6 +235,8 @@ func (r *ReconcileWebSphereLiberty) Reconcile(ctx context.Context, request ctrl.
 	if imageReferenceOld != instance.Status.ImageReference {
 		reqLogger.Info("Updating status.imageReference", "status.imageReference", instance.Status.ImageReference)
 		err = r.UpdateStatus(instance)
+		// Trigger a new Semeru Cloud Compiler generation
```

Comment:
The above update status could fail. So it's better to invoke createNewSemeruGeneration prior to status update call. So anytime the image is updated, the generation counter will be increased and both will be stored in status.

Reply:
Done